### PR TITLE
ENH: Add dtype to read sql to be consistent with read_sql_query

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Came across this, seemed odd to me that one supports it but not the other

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -659,6 +668,7 @@ def read_sql(
                 parse_dates=parse_dates,
                 chunksize=chunksize,
                 use_nullable_dtypes=use_nullable_dtypes,
+                dtype=dtype,
```
Comment (mroeschke) [inline pandas/io/sql.py:671]: How does this interact with `use_nullable_dtypes` when `dtype` is not a nullable type?

Comment (phofl) [reply pandas/io/sql.py:671]: We should add tests for this :)

Right now we are casting to nullable and then apply astype (that's how it was implemented before as well). But I don't think that we have any tests covering this

Comment (mroeschke) [reply pandas/io/sql.py:672]: Yeah would be good to add tests for this case. IIRC `read_csv` or a similar reader had a similar thing. This should hopefully operate similarly

Comment (phofl) [reply pandas/io/sql.py:672]: added

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**